src/advanced_inv.cpp

```diff
@@ -1362,7 +1383,9 @@ void advanced_inventory::display()
             if( !query_destination( destarea ) ) {
                 continue;
             }
-            if( squares[srcarea].is_same( squares[destarea] ) ) {
+            if( squares[srcarea].is_same( squares[destarea] ) && 
```

Comment:
This part (the block starting at line 1356|1377 with `action == "MOVE_SINGLE_ITEM") has one big problem:
 the source and target are only defined as `aim_location`, which basically excludes vehicle-to-vehicle transfers (`srcarea == AIM_VEHICLE`, `destarea == AIM_VEHICLE`, huh?)

I see at least two possible solutions:
- Move most of the code into the `advanced_inventory_pane` class and use the members of that class to identify where to move from/to (use `area` *and* `veh_area` and maybe more).
- Add more aim_locations: one for each direction (N, S, W, ...) and ALL, but for vehicle storage, they would act like AIM_VEHICLE but with specific locations. (AIM_VEHICLE does not have a location, doesn't it?)